src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,203 @@
+        String result = "";
+        try {
+            if (separateCommand.length > 2 || Integer.parseInt(separateCommand[1]) > tasks.getAll().size()) {
+                throw new DukeException(" OOPS!!! Invalid number");
+            }
+            int taskNumber = Integer.parseInt(separateCommand[1]);
+            Task task = tasks.getTask(taskNumber - 1);
+            if (command.startsWith("delete")) {
+                result += " Noted. I've removed this task:\n";
+                result += "   " + task.toString() + "\n";
+                tasks.removeTask(taskNumber - 1);
+            }
+            Ui.showMessage(result);
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+            throw new DukeException(" OOPS!!! Invalid number");
+        }
+        return result;
+    }
+
+    /**
+     * Executes the 'find' command to search for tasks containing a specific keyword.
+     *
+     * @param command The user command to execute ('find').
+     * @param tasks   The TaskList object to search for matching tasks.
+     */
+    public static String executeFind(String command, TaskList tasks) {
+        String result = "";
+        String keyword = command.substring(5).trim();
+        List<Task> matchingTasks = tasks.findTasks(keyword);
+        result += " Here are the matching tasks in your list:\n";
+        for (int i = 0; i < matchingTasks.size(); i++) {
+            result += " " + (i + 1) + "." + matchingTasks.get(i).toString() + "\n";
+        }
+        Ui.showMessage(result);
+        return result;
+    }
+
+    /**
+     * Executes the specified command and performs the corresponding action.
+     *
+     * @param command The user command to be executed.
+     * @param tasks   The TaskList object to manage tasks.
+     * @throws DukeException If there is an issue or error encountered during command execution.
+     */
+    public static String executeCommand(String command, TaskList tasks) {
+        String result = "";
+        try {
+            String[] separateCommand = command.split(" ");
+            if (command.equals("list")) {
+                result += Parser.executeList(tasks);
+            } else if (command.startsWith("mark") || command.startsWith("unmark")) {
+                result += Parser.executeMarkUnmark(command, tasks, separateCommand);
+            } else if (command.startsWith("todo") || command.startsWith("deadline") || command.startsWith("event")) {
+                result += Parser.executeCreate(command, tasks);
+            } else if (command.startsWith("delete")) {
+                result += Parser.executeDelete(command, tasks, separateCommand);
+            } else if (command.startsWith(("find"))) {
+                result += Parser.executeFind(command, tasks);
+            } else {
+                result += " OOPS!!! I'm sorry, but I don't know what that means :-(";
```

Comment:
I like how this method is still easy to read even though there are a lot of if else checks to be done

Reply:
thank you!